code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_timestop.dm

```diff
@@ -0,0 +1,49 @@
+/datum/artifact_effect/timestop
+	effecttype = "timestop"
+
+	var/mob/caster
+	var/spell/aoe_turf/fall/fall
+	var/stop_length
+	copy_for_battery = list("stop_length")
+
+/datum/artifact_effect/timestop/New()
+	..()
+	effect = pick(EFFECT_TOUCH, EFFECT_PULSE)
+	caster = new(src)
+	caster.invisibility = 101
+	caster.density = 0
+	caster.anchored = 1
+	caster.flags = INVULNERABLE
+	caster.add_spell(new /spell/aoe_turf/fall)
```

Comment:
Why do you create the spell only to locate() it later? You should do `fall = new /spell/aoe_turf/fall`, and then `caster.add_spell(fall)`